### PR TITLE
feat: add prefix option to all created resources

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description

Issue:https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/issues/455

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/standard/variables.tf [hunk 1]
```diff
@@ -25,14 +25,20 @@ variable "region" {
   description = "The region in which to provision KMS resources. If using existing KMS, set this to the region in which it is provisioned in."
```
Comment (MatthewLemmond) [inline solutions/standard/variables.tf:25]: This description should also have a similar comment as the one added to the description of `key_protect_instance_name` noting the optional addition of the prefix if provided

**Review by MatthewLemmond — CHANGES_REQUESTED** [review]
Just the comment about the description for resource group name, otherwise looks good

File: solutions/standard/variables.tf (hunk 1 above)
Comment (ocofaigh) [reply solutions/standard/variables.tf:25]: i guess your talking about the `resource_group_name` descrption

Comment (Khuzaima05) [reply solutions/standard/variables.tf:25]: Done, updated the resource_group_name description.

File: solutions/standard/variables.tf [hunk 2]
```diff
@@ -25,14 +25,20 @@ variable "region" {
...
+
 ########################################################################################################################
 # Key Protect instance variables
 ########################################################################################################################
 
 variable "key_protect_instance_name" {
   type        = string
   default     = "base-security-services-kms"
-  description = "The name to give the Key Protect instance that will be provisioned by this solution. Only used if not supplying an existing KMS instance."
+  description = "The name to give the Key Protect instance that will be provisioned by this solution. Only used if not supplying an existing KMS instance. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value"
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:41]: same as https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/pull/460/files#r1569179411

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Any changes to variables for the DA will need to also be made in ibm_catalog.json since we are duplicating them there so we can order them in catalog. Please add `prefix` to the [configuration](https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/blob/d6bfbb7ebaf803310608a8798bac1d7295c4683d/ibm_catalog.json#L63) array in the json. I think it can be come after `region` just like it does in the variables.tf file itself

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**